File: app/src/main/java/com/seniorzhai/dagger2sample/model/Gank.java
package com.seniorzhai.dagger2sample.model;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by zhai on 16/5/23.
 */

public class Gank implements Serializable {
    protected long id;
    public String objectId;

    public boolean used;
    public String type;
    public String url;
    public String who;
    public String desc;
    public Date updatedAt;
    public Date createdAt;
    public Date publishedAt;

    public boolean is妹子() {
        return "福利".equals(type);
    }

    @Override
    public Gank clone() {
        Gank gank = null;
        try {
            gank = (Gank) super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }
        return gank;
    }
}

File: app/src/main/java/com/seniorzhai/dagger2sample/model/Gank.kt
package com.seniorzhai.dagger2sample.model

import java.io.Serializable
import java.util.*

/**
 * Created by zhai on 16/5/23.
 */

class Gank : Serializable {
    protected var id: Long = 0
    var objectId: String? = null

    var used: Boolean = false
    var type: String? = null
    var url: String? = null
    var who: String? = null
    var desc: String? = null
    var updatedAt: Date? = null
    var createdAt: Date? = null
    var publishedAt: Date? = null

    val is妹子: Boolean
        get() = "福利" == type
}
